src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -510,14 +541,23 @@ private IGraphType ConvertTypeReference(INamedType parentType, IGraphType type)
             }
 
             var reference = type as GraphQLTypeReference;
-            var result = reference == null ? type : this[reference.TypeName];
+            if (reference != null)
+            {
+                type = this[reference.TypeName];
+                if (type == null)
+                {
+                    type = _builtInScalars.Values.FirstOrDefault(t => t.Name == reference.TypeName);
+                    if (type != null)
+                        this[type.Name] = type;
```

Comment:
Example - discovering `ID` built-in type during schema first for 
```graphql
type Post {
                    id: ID!
                    title: String!
                }

                type Query {
                    post(id: ID!): Post
                }
```

Reply:
I don't use schema-first so I'll trust you here :-)

Reply:
Wouldn't this also need to check the `_builtInCustomScalars`?  Seems like that would be the case everywhere, it would need to check both wouldn't it?

Reply:
You are right. Fixed.